www/updateUes.php

```diff
@@ -36,91 +57,217 @@
     $perunUserId = $claims['data']['perunUserId'];
     $id = $claims['id'];
 } catch (Exception $ex) {
-    Logger::error('Perun.updateUes: An error occurred when the token was verifying.');
+    Logger::error(CLASS_PREFIX . 'An error occurred when the token was verifying.');
     http_response_code(400);
     exit;
 }
 
 try {
-    if (empty($attributesFromIdP['sourceIdPEntityID'][0])) {
-        throw new Exception('perun/www/updateUes.php: Invalid attributes from Idp - sourceIdPEntityID is empty');
+    $config = Configuration::getConfig(CONFIG_FILE_NAME);
+    if (null === $config) {
+        throw new Exception(
+            CLASS_PREFIX . 'Configuration not set - config file \'' . CONFIG_FILE_NAME . '\' is missing'
+        );
     }
 
-    if (empty($attributesFromIdP['sourceIdPEppn'][0])) {
-        throw new Exception('perun/www/updateUes.php: Invalid attributes from Idp - sourceIdPEppn is empty');
+    $config = $config->getArray(CONFIG_SECTION, null);
+    if (null === $config) {
+        throw new Exception(
+            CLASS_PREFIX . 'Configuration not set - section \'' . CONFIG_SECTION . '\' not found in config'
+        );
     }
 
-    $userExtSource = $adapter->getUserExtSource(
-        $attributesFromIdP['sourceIdPEntityID'][0],
-        $attributesFromIdP['sourceIdPEppn'][0]
-    );
-    if (null === $userExtSource) {
+    $userIdentifiers = [];
+    if (is_array($config[USER_IDENTIFIERS]) && !empty($config[USER_IDENTIFIERS])) {
+        $uids = $config[USER_IDENTIFIERS];
+    } else {
+        throw new Exception(CLASS_PREFIX . 'Config option \'' . USER_IDENTIFIERS . '\' is not set or is empty/invalid');
+    }
+
+    if (empty($attributesFromIdP[SOURCE_IDP_ENTITY_ID][0])) {
```

Comment:
Would be better if the attribute key is configurable. Otherwise, we create a strong dependency on the filters that have to be run before the updateUes, which is OK, but we also do not give the flexibility to use the naming of attributes the person who has configured it wants to use.